### PR TITLE
Add query multiple tables

### PR DESCRIPTION
**Pull request by KAYEMBAJONATHAN**
- Add Create a table named owners
- Add Create a table named species
- Add Modify animals table
- Add Modify your inserted animals 
- Abb Write queries using JOIN to answer different questions

### CODE REVIEW

**Comment by KAYEMBAJONATHAN** [comment]
> * Add Create a table named owners
> * Add Create a table named species
> * Add Modify animals table
> * Add Modify your inserted animals
> * Abb Write queries using JOIN to answer different questions

Screenshot 2023-06-15 203027 owenrs
Screenshot 2023-06-15 203154  species
Screenshot 2023-06-15 203731 update
Screenshot 2023-06-15 203848 set
Screenshot 2023-06-15 204509 names
Screenshot 2023-06-15 204720  a n w
Screenshot 2023-06-15 204944 f a
Screenshot 2023-06-15 205043 s m
Screenshot 2023-06-15 205206 d f
Screenshot 2023-06-15 205329 ES
Screenshot 2023-06-15 205537 o f

File: schema.sql [hunk 1]
```diff
@@ -7,6 +7,27 @@ CREATE TABLE animals (
...
     weight_kg DECIMAL
+    species_id INT,
+    owners_id INT,
+    PRIMARY KEY (id),
+    CONSTRAINT fk_species
+    FOREIGN KEY(species_id) 
+	REFERENCES species(id),
+    CONSTRAINT fk_owners
+    FOREIGN KEY(owners_id) 
+	REFERENCES owners(id)
```
Comment (laiifuu) [inline schema.sql]: When Making changes to a table, you should use the `ALTER TABLE` command. This way you wouldn't have to delete the table and re-create again. You did write altering commands in the `data.sql` file, so I suggest you revert this to how it was before.

File: data.sql [hunk 2]
```diff
@@ -3,11 +3,36 @@ INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg)
...
+        (9,'Boarmon',  '2005-06-07', 7, true , 20.4),
+        (10,'Blossom', '1998-10-13', 3 , true , 17),
+        (11,'Ditto', '2022-05-14' , 4 , true , 22);
+
+ALTER TABLE animals ADD PRIMARY KEY (id);
+ALTER TABLE animals DROP COLUMN species;
+ALTER TABLE animals ADD COLUMN species_id INT;
+ALTER TABLE animals ADD COLUMN owners_id INT;
+ALTER TABLE animals ADD CONSTRAINT fk_species FOREIGN KEY (species_id) REFERENCES species(id);
+ALTER TABLE animals ADD CONSTRAINT fk_owners FOREIGN KEY (owners_id) REFERENCES owners(id);
```
Comment (laiifuu) [inline data.sql]: These commands modify the database's schema, I would suggest you place them in `schema.sql` file which is the appropriate file for them.

File: data.sql [hunk 3]
```diff
@@ -3,11 +3,36 @@ INSERT INTO animals (id,name,date_of_birth,escape_attempts,neutered,weight_kg)
...
+INSERT INTO owners (full_name, age) VALUES ('Sam Smith', 34);
+INSERT INTO owners (full_name, age) VALUES ('Jennifer Orwell', 19);
+INSERT INTO owners (full_name, age) VALUES ('Bob', 45);
+INSERT INTO owners (full_name, age) VALUES ('Melody Pond', 77);
+INSERT INTO owners (full_name, age) VALUES ('Dean Winchester', 14);
+INSERT INTO owners (full_name, age) VALUES ('Jodie Whittaker', 38);
+INSERT INTO species (name) VALUES ('Pokemon');
+INSERT INTO species (name) VALUES ('Digimon');
+UPDATE animals SET species_id = 2 WHERE name LIKE '%mon';
+UPDATE animals SET species_id = 1 WHERE species_id ISNULL;
```
Comment (laiifuu) [inline data.sql]: Kindly fix this typo: `ISNULL => IS NULL`

File: queries.sql [hunk 4]
```diff
@@ -9,26 +9,39 @@ SELECT * FROM animals WHERE (weight_kg >= 10.4 AND weight_kg <= 17.3);
...
 SELECT COUNT(*) FROM animals;
 SELECT COUNT(*) FROM animals WHERE (escape_attempts = 0);
 SELECT AVG(weight_kg)  FROM animals;
 SELECT neutered,AVG(escape_attempts) AS escape FROM animals GROUP BY (neutered) ORDER BY escape DESC LIMIT 1;
 SELECT MIN(weight_kg),MAX(weight_kg), species FROM animals GROUP BY species;
-SELECT species,AVG(escape_attempts) AS escape FROM animals WHERE date_of_birth BETWEEN  '1990-01-01' AND '2000-01-01' GROUP BY (species) ;
\ No newline at end of file
+SELECT species,AVG(escape_attempts) AS escape FROM animals WHERE date_of_birth BETWEEN  '1990-01-01' AND '2000-01-01' GROUP BY (species);
+
+SELECT animals.name, owners.full_name FROM animals JOIN owners ON animals.owners_id=owners.id WHERE full_name = 'Melody Pond';
```
Comment (laiifuu) [inline queries.sql:41]: This query should show the following animal names: `Charmander`, `Squirtle`, and `Blossom`. But according to your screenshot, the result is `Angemon` and `Boarmon` which belong to `Dean Winchester`. It's possible there was a mix up when setting the owner's id when updating the `animals` table. Kindly fix this issue.

File: queries.sql [hunk 5]
```diff
@@ -9,26 +9,39 @@ SELECT * FROM animals WHERE (weight_kg >= 10.4 AND weight_kg <= 17.3);
...
+SELECT animals.name, species.name FROM animals JOIN species ON animals.species_id=species.id WHERE species.name = 'Pokemon';
+SELECT owners.full_name, animals.name AS animal_name FROM owners LEFT JOIN animals ON owners.id=animals.owners_id;
+SELECT species.name, COUNT(*) AS species_number FROM species JOIN animals ON species.id = animals.species_id GROUP BY species.name;
+SELECT name AS digimon_animals, owners.full_name FROM (SELECT animals.name, animals.owners_id FROM animals JOIN SPECIES ON species_id = species.id WHERE species.name LIKE 'Digimon') AS all_digimons JOIN owners ON all_digimons.owners_id = owners.id WHERE owners.full_name LIKE 'Jennifer Orwell';
```
Comment (laiifuu) [inline queries.sql:45]: This query should show the following animal name: `Gabumon`. But according to your screenshot, the result is `Devimon` and `Plantmon` which belong to Bob. It's possible there was a mix up when setting the owner's id when updating the animals table. Kindly fix this issue.

File: queries.sql [hunk 6]
```diff
@@ -9,26 +9,39 @@ SELECT * FROM animals WHERE (weight_kg >= 10.4 AND weight_kg <= 17.3);
...
+SELECT animals.name FROM animals JOIN owners ON animals.owners_id=owners.id WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts<1;
+SELECT owners.full_name, COUNT(animals.name) AS all_animals from animals JOIN owners ON animals.owners_id = owners.id GROUP BY owners.full_name ORDER BY all_animals DESC LIMIT 1;
```
Comment (laiifuu) [inline queries.sql:47]: This query should show Melody Pond with 3 animals, but according to your screenshot, the result is Bob with 3 animals. It's possible there was a mix up when setting the owner's id when updating the animals table. Kindly fix this issue.

**Review by laiifuu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: 

## Hi @KAYEMBAJONATHAN :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
so-close

## Highlights
- [x] Nice use of the JOIN command for the queries :+1: 

## Required Changes ♻️
_Check the comments under the review :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KAYEMBAJONATHAN** [comment]
See below the screen for what I have fixed

Screenshot 2023-06-16 095635 fixed1
Screenshot 2023-06-16 095729 fixed 2

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @KAYEMBAJONATHAN, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**